internal/helm/helm.go

```diff
@@ -80,36 +83,37 @@ func NewClient(kubeConfig []byte, log debugLog) (*Client, error) {
 type Options struct {
 	Conformance      bool
 	HelmWaitMode     WaitMode
+	DeployCSIDriver  bool
 	AllowDestructive bool
 	Force            bool
 	ApplyTimeout     time.Duration
 }
 
 // PrepareApply loads the charts and returns the executor to apply them.
-// TODO(elchead): remove validK8sVersion by putting ValidK8sVersion into config.Config, see AB#3374.
 func (h Client) PrepareApply(
-	conf *config.Config, stateFile *state.State,
-	flags Options, serviceAccURI string, masterSecret uri.MasterSecret,
+	csp cloudprovider.Provider, attestationVariant variant.Variant, k8sVersion versions.ValidK8sVersion,
+	microserviceVersion semver.Semver, stateFile *state.State, flags Options, serviceAccURI string,
+	masterSecret uri.MasterSecret, openStackCfg *config.OpenStackConfig,
 ) (Applier, bool, error) {
-	releases, err := h.loadReleases(conf, masterSecret, stateFile, flags, serviceAccURI)
+	releases, err := h.loadReleases(csp, attestationVariant, k8sVersion, masterSecret, stateFile, flags, serviceAccURI, openStackCfg)
```

Comment:
Should we add these two checks right here because its the public entrypoint for the lib?
- `if csp == cloudprovider.OpenStack && openStackCfg == nil { return err; }`
- `if csp != cloudprovider.OpenStack && openStackCfg != nil { return err; }`